### PR TITLE
KYLIN-4114 Provided a self-contained docker image for Kylin

### PR DESCRIPTION
**Pull request by weibin0516**

### CODE REVIEW

**Comment by hit-lacus** [comment]
Thanks @weibin0516 , this looks cool, here a list for what I want to check. 

### For PoC
- [ ] MapReduce Cubing
- [ ] Spark Cubing
- [ ] Segment Merge
- [ ] NRT Streaming
- [ ] Realtime OLAP

### For Developer
- [ ] Package from source
- [ ] Integration test


### Other Comments
- I guess we may check if we could upload image to docker hub, thus user may simplely run it in one command. (For PoC purpose.)
- Maybe we could build image in hierarchy way, thus we can switch between different hadoop distribution.
- Make sure all basic function works well.

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+# setup hadoop
+RUN wget https://archive.apache.org/dist/hadoop/core/hadoop-2.7.0/hadoop-2.7.0.tar.gz
+RUN tar -zxvf /home/admin/hadoop-$HADOOP_VERSION.tar.gz
+RUN rm -f /home/admin/hadoop-$HADOOP_VERSION.tar.gz
+
+COPY conf/hadoop/* $HADOOP_CONF/
+RUN mkdir -p /data/hadoop
+
+# setup hbase
+RUN wget https://archive.apache.org/dist/hbase/1.1.2/hbase-1.1.2-bin.tar.gz
```
Comment (hit-lacus) [inline docker/Dockerfile]: Forget to use `$HBASE_VERSION` here?

File: docker/use-guide.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+When building image, the latest kylin source code will be copied to `/home/admin/kylin_sourcecode`.
+
+Execute the following commands to do package:
+```
+cd /home/admin/kylin_sourcecode
+build/script/package.sh
+```
+After the package is complete, use the package in `/home/admin/kylin_sourcecode/dist` to start the Kylin service.
+
+You can also run CI inside the container:
```
Comment (hit-lacus) [inline docker/use-guide.md]: If Integration test can not pass currently, please remove this, thank you.

Comment (weibin0516) [reply docker/use-guide.md]: fixed

File: docker/Dockerfile (hunk 1 above)
Comment (weibin0516) [reply docker/Dockerfile]: fixed

**Comment by weibin0516** [comment]
@hit-lacus , thanks for review, all functions except `Integration test` work fine work fine. In my opinion, Integration test is not necessarily needed, may be i will solve it in next pr.

**Comment by hit-lacus** [comment]
Hi, @weibin0516 

Did you update your PR? I cannot find `kylin.sh start` & `kylin.sh streaming start` anywhere.

File: docker/run_container.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+docker run -d \
+-m 8G \
+-p 7070:7070 \
+-p 8088:8088 \
+-p 50070:50070 \
+-p 8032:8032 \
+-p 8042:8042 \
+-p 60010:60010 \
+apache-kylin-standalone
```
Comment (hit-lacus) [inline docker/run_container.sh]: Should be `codingforfun/apache-kylin-standalone` instead.

**Comment by hit-lacus** [comment]
## Test Env 

- MacOS 10.14.4 
- Docker Desktop for Mac (Docker Engine: 18.09.2)

#### Start standalone docker container 
```sh
docker run -d --rm \
-p 7070:7070 \
-p 8088:8088 \
-p 50070:50070 \
-p 8032:8032 \
-p 8042:8042 \
-p 60010:60010 \
codingforfun/apache-kylin-standalone
```

#### Attach to container
``` sh
docker exec -it ${CONTAINER_ID} /bin/bash
```


#### Building Cube 

image


image


#### Near Realtime Streaming

image

#### Realtime OLAP

image

image

### Known issue
image

Nodemanager may be unhealthy because of lack of disk space.
This may be fixed by `docker image prune -h` to reduce needless disk space. 

Container cannot restart successfully because `entrypoint.sh`.

Realtime OLAP building job cubing job throw NPE for some reasons(to be analysed).

**Comment by weibin0516** [comment]
@shaofengshi , thanks very much.

File: docker/use-guide.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+* Zookeeper 3.4.6
+* Kafka 1.1.1
+* Mysql
+* Maven 3.6.1
+
+## For user to try Kylin
+The image has been pushed to the docker hub without having to build locally.
+Execute the following command to pull image: 
+```
+docker pull codingforfun/apache-kylin-standalone
```
Comment (hit-lacus) [inline docker/use-guide.md]: Could you please update this?

**Pull request merged**